editor/types/globalSlice.d.ts

```diff
@@ -0,0 +1,74 @@
+interface Fiber {
+  color: string;
+  alpha: number; //brightness
+}
+
+type El = number;
+
+interface LED {
+  src: string;
+  alpha: number;
+}
+
+interface ControlMapStatus {
+  [index: string]: Fiber | El | LED;
+}
+
+interface ControlMapElement {
+  start: number; //frame's start time
+  status: ControlMapStatus;
+  fade: boolean; // if this frame fades to the next
+}
+interface timeDataType {
+  from: string; // update from what component
+  time: number; // time
+  controlFrame: number; // control frame's index
+  posFrame: number; // positions' index
+}
+interface coordinates {
+  x: number;
+  y: number;
+  z: number;
+}
+
+interface positionType {
+  [index: string]: coordinates; //dancer corresponds to his/her position
+}
+
+interface posRecordElement {
+  pos: positionType;
+  start: number; //start time
+}
+
+interface lightPresetsElement {
+  name: string; // ID named by user
+  status: ControlMapStatus;
+}
+interface posPresetsElement {
+  name: string;
+  pos: positionType;
+}
+type LightPresetsType = lightPresetsElement[];
+type posPresetsType = posPresetsElement[];
+export interface posRecordType {
```

Comment:
posRecordType is an array of posRecordElement.
`export type posRecordType = Array<posRecordElement>;`